data-prepper-core/src/main/java/org/opensearch/dataprepper/pipeline/router/RouterCopyRecordStrategy.java

```diff
@@ -48,16 +54,36 @@ public <C> RouterCopyRecordStrategy(final EventFactory eventFactory, final Ackno
         }
     }
 
+    Set<Record> getReferencedRecords() {
+        return referencedRecords;
+    }
+
+    private void acquireEventReference(Record record) {
+        if (acknowledgementSetManager == null || record.getData() == null) {
+            return;
+        }
+        if (referencedRecords.contains(record) || ((routedRecords != null) && routedRecords.contains(record))) {
+            EventHandle eventHandle = ((JacksonEvent)record.getData()).getEventHandle();
+            if (eventHandle != null) {
+                acknowledgementSetManager.acquireEventReference(eventHandle);
+            }
+        } else if (!referencedRecords.contains(record)) {
+            referencedRecords.add(record);
+        }
+    }
     @Override
     public Record getRecord(Record record) {
         if (routedRecords == null) {
+            acquireEventReference(record);
             return record;
         }
         if (!routedRecords.contains(record)) {
+            acquireEventReference(record);
             routedRecords.add(record);
             return record;
         }
         if (record.getData() instanceof JacksonSpan) {
+            // Not supporting acknowledgements for Span initially
```

Comment:
We should call this out in documentation as well. (I assume you will but this comment was a good one that I forgot about). Perhaps we should create a github issue to track this?